Mathlib/Algebra/Module/Presentation/Differentials.lean

```diff
@@ -0,0 +1,164 @@
+Indeed, there is commutative diagram with a surjective map
+`hom₁ : (pres.rels →₀ S) →ₗ[S] pres.toExtension.Cotangent` on the left and
+bijections on the middle and on the right. Then, the exactness of the first
+sequence shall follow from the exactness of the second which is
+`Algebra.Extension.exact_cotangentComplex_toKaehler`. -/
+
+/-- Same as `comm₂₃` below, but here we have not yet constructed `differentialsSolution`. -/
+lemma comm₂₃' : pres.toExtension.toKaehler.comp pres.cotangentSpaceBasis.repr.symm.toLinearMap =
+    Finsupp.linearCombination S (fun g ↦ D _ _ (pres.val g)) := by
+  ext g
+  dsimp
+  rw [Basis.repr_symm_apply, Finsupp.linearCombination_single,
+    Finsupp.linearCombination_single, one_smul, one_smul,
+    Generators.cotangentSpaceBasis_apply, mapBaseChange_tmul, one_smul]
+  simp
+
+/-- The canonical map `(pres.rels →₀ S) →ₗ[S] pres.toExtension.Cotangent`. -/
+noncomputable def hom₁ : (pres.rels →₀ S) →ₗ[S] pres.toExtension.Cotangent :=
+  Finsupp.linearCombination S (fun r ↦ Extension.Cotangent.mk ⟨pres.relation r, by simp⟩)
+
+lemma hom₁_single (r : pres.rels) :
+    hom₁ pres (Finsupp.single r 1) = Extension.Cotangent.mk ⟨pres.relation r, by simp⟩ := by
+  simp [hom₁]
+
+lemma surjective_hom₁ : Function.Surjective (hom₁ pres) := by
+  let φ : (pres.rels →₀ S) →ₗ[pres.Ring] pres.toExtension.Cotangent :=
+    { toFun := hom₁ pres
+      map_add' := by simp
+      map_smul' := by simp }
+  change Function.Surjective φ
+  have h₁ := Algebra.Extension.Cotangent.mk_surjective (P := pres.toExtension)
+  have h₂ : Submodule.span pres.Ring
+      (Set.range (fun r ↦ (⟨pres.relation r, by simp⟩ : pres.ker))) = ⊤ := by
+    refine Submodule.map_injective_of_injective (f := Submodule.subtype pres.ker)
+      Subtype.coe_injective ?_
+    rw [Submodule.map_top, Submodule.range_subtype, Submodule.map_span,
+      Submodule.coe_subtype, Ideal.submodule_span_eq]
+    simp only [← pres.span_range_relation_eq_ker]
+    congr
+    aesop
+  rw [← LinearMap.range_eq_top] at h₁ ⊢
+  rw [← top_le_iff, ← h₁, LinearMap.range_eq_map, ← h₂]
+  dsimp
+  rw [Submodule.map_span_le]
+  rintro _ ⟨r, rfl⟩
+  simp only [LinearMap.mem_range]
+  refine ⟨Finsupp.single r 1, ?_⟩
+  simp only [LinearMap.coe_mk, AddHom.coe_mk, hom₁_single, φ]
+  rfl
+
+lemma comm₁₂_single (r : pres.rels) :
+    pres.toExtension.cotangentComplex (hom₁ pres (Finsupp.single r 1)) =
+      pres.cotangentSpaceBasis.repr.symm ((differentialsRelations pres).relation r) := by
+  simp only [hom₁, Finsupp.linearCombination_single, one_smul, differentialsRelations,
+    Basis.repr_symm_apply, Extension.cotangentComplex_mk]
+  exact pres.cotangentSpaceBasis.repr.injective (by ext; simp)
+
+lemma comm₁₂ : pres.toExtension.cotangentComplex.comp (hom₁ pres) =
+    pres.cotangentSpaceBasis.repr.symm.comp (differentialsRelations pres).map := by
+  ext r
+  dsimp
+  rw [comm₁₂_single]
+  erw [Module.Relations.map_single]
```

Comment:
```suggestion
  simp_rw [← differentialsRelations_G, ← differentialsRelations_R, Module.Relations.map_single]
```
The alternative to this is removing `@[simp]` from `differentialsRelations_{G,R}` and replace occurrences of `pres.vars` and `pres.rels` with `differentialsRelations.G` and `differentialsRelations.R` respectively.

Reply:
Removing the `simps` attribute is not an option here, but I have found another way to remove the `erw` (this type of `erw` will be solved by the expected `dsimp%` elaborator).